service/security/test/src/test/resources/features/GroupServiceUnitTests.feature

```diff
@@ -0,0 +1,387 @@
+  Create an Access Group with valid name. Edit the group name to contain valid special symbols: "_", "-" and " ". Try to search for it - it should have been changed.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description ""
+    When I update the group name to "New Access_Group-ag"
+    And I search for the group with name "New Access_Group-ag"
+    Then I find the group with name "New Access_Group-ag"
+    And The group was correctly updated
+    And No exception was thrown
+
+  Scenario: Editing Access Group name to name with invalid special symbols in name
+  Create an Access Group with valid name. Edit the group name to name with invalid special symbols in name, all except "_", "-" and " ". Use one symbol every time.
+  Kapua should return an error.
+    Given I create the group with name "NewAccessGroup" and description ""
+    Given I expect the exception "KapuaIllegalArgumentException" with the text "*"
+    When I update the group name to "NewAccessGroup@"
+    Then An exception was thrown
+
+  Scenario: Editing Access Group name to empty name
+  Create an Access Group with valid name. Edit the group name to empty name.
+  Kapua should return an error.
+    Given I create the group with name "NewAccessGroup" and description ""
+    Given I expect the exception "KapuaIllegalNullArgumentException" with the text "*"
+    When I update the group name to " "
+    Then An exception was thrown
+
+  Scenario: Editing Access Group name to name that contains numbers
+  Create an Access Group with valid name. Edit the group name to contain numbers. Try to search for it - it should have been changed.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description ""
+    When I update the group name to "Group123"
+    And I search for the group with name "Group123"
+    Then I find the group with name "Group123"
+    And The group was correctly updated
+    And No exception was thrown
+
+  Scenario: Editing Access Group name to name that contains only numbers
+  Create an Access Group with valid name. Edit the group name to contain only numbers. Try to search for it - it should have been changed.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description ""
+    When I update the group name to "1234"
+    And I search for the group with name "1234"
+    Then I find the group with name "1234"
+    And The group was correctly updated
+    And No exception was thrown
+
+  Scenario: Editing Access Group description to unique one
+  Create an Access Group with unique name and with unique description. Once created, search for that description - it should have been changed.
+  Kapua should not return any errors.
+    Given I create the group with name "NewAccessGroup" and description "description"
+    When I update the group description to "NewDescription"
```

Comment:
Please change this step to: "When I update description of group "newAccessGroup" from "description" to "NewDescription"" so the implementation receives two parameters instead of one. This way the step is more clear and easier to understand.